asistente_ladm_col/lib/db/mssql_connector.py

```diff
@@ -0,0 +1,526 @@
+        uri = []
+
+        uri += ['DRIVER={{{}}}'.format(dict_conn['db_odbc_driver'])]
+        host = dict_conn['host'] or self._DEFAULT_VALUES['host']
+        if dict_conn['instance']:
+            host += '\\' + dict_conn['instance']
+        if dict_conn['port']:
+            host += ',' + dict_conn['port']
+
+        uri += ['SERVER={}'.format(host)]
+        if dict_conn['database'] and level == 1:
+            uri += ['DATABASE={}'.format(dict_conn['database'])]
+        uri += ['UID={}'.format(dict_conn['username'])]
+        uri += ['PWD={}'.format(dict_conn['password'])]
+
+        return ';'.join(uri)
+
+    def get_ili2db_version(self):
+        res, msg = self.check_and_fix_connection()
+        if not res:
+            return res, msg
+
+        # Borrowed from Model Baker
+        cur = self.conn.cursor()
+        cur.execute("""SELECT count(COLUMN_NAME)
+                       FROM information_schema.columns
+                       WHERE table_schema = '{schema}'
+                       AND(table_name='t_ili2db_attrname' OR table_name='t_ili2db_model' )
+                       AND(column_name='owner' OR column_name = 'file' )
+                    """.format(schema=self.schema))
+
+        columns_count = cur.fetchone()[0]
+
+        if columns_count > 0:
+            return 3
+        else:
+            return 4
+
+    def check_and_fix_connection(self):
+        if self.conn is None:
+            res, code, msg = self.test_connection()
+            if not res:
+                return res, msg
+
+        return True, ''
+
+    def get_logic_validation_queries(self):
+        # return variables from logic_validation_queries for mssql
+        pass
+
+    def _test_connection_to_schema(self, user_level):
+        # TODO # is 'dbo' database valid?  self._dict_conn_params['schema'] == 'dbo':
+        if not self._dict_conn_params['schema']:
+            return False, EnumTestConnectionMsg.SCHEMA_NOT_FOUND, QCoreApplication.translate("MSSQLConnector",
+                                                                                             "You should first select a schema.")
+        if not self._schema_exists():
+            return False, EnumTestConnectionMsg.SCHEMA_NOT_FOUND, QCoreApplication.translate("MSSQLConnector",
+                                                                                             "The schema '{}' does not exist in the database!").format(
+                self.schema)
+        # TODO Test schema permissions (*)
+
+        return True, EnumTestConnectionMsg.CONNECTION_TO_SCHEMA_SUCCESSFUL, QCoreApplication.translate("PGConnector",
+                                                                                                       "Connection to the database schema was successful.")
+
+    def _test_connection_to_server(self):
+        # why is this method called necessary?
+        uri = self.get_connection_uri(self._dict_conn_params, 0)
```

Comment:
Because sometimes we need to make sure we can connect to a server level (e.g., to create DBs).
(Remove comment)